### PR TITLE
Added `verify-book-links` make target

### LINKED ISSUE #3176: Add verify-book-links to Makefile target
There are a few Makefile targets that are used in other cluster api projects that will be great to add to CAPA. 

One of them is `verify-book-links` which checks if book links are correct

Refer to [cluster-api](https://github.com/kubernetes-sigs/cluster-api/blob/main/Makefile#L365-L367) for implementation details.

### PR DESCRIPTION
**Pull request by zeborg**
Signed-off-by: Abhinav Sinha <abhinav@nirmata.com>

**What type of PR is this?**
`/kind feature`

**What this PR does / why we need it**:
* Added `verify-book-links` make target
* Upgraded image `maven:3-jdk-14` to `maven:3-openjdk-17-slim` in `hack/tools/plantuml.Dockerfile` for multi-arch _(amd64 and arm64)_ support
* Updated `docs/proposal/20200506-single-controller-multitenancy-flow.svg` to the newly generated one

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3176 

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Added `verify-book-links` make target
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -615,6 +615,11 @@ verify-conversions: $(CONVERSION_VERIFIER)  ## Verifies expected API conversion
 verify-shellcheck: ## Verify shell files
 	./hack/verify-shellcheck.sh
 
+.PHONY: verify-book-links
+verify-book-links: ## Verify book links
+	echo verification of book links initiated
```
Comment (sedefsavas) [inline Makefile:620]:
```suggestion
```

**Comment by zeborg** [comment]
Hi @sedefsavas, looking at the failed logs it seems like in order for this test to succeed we also need to install Docker in the `verify-book-links` CI job. I'd like to know if I should implement this change as well. Thanks!

**Comment by zeborg** [comment]
Update:

Hi @sedefsavas, I tried to check if Docker was present in the CI environment, and it seems like it is already installed but the docker daemon is not running. I'm not sure about what the underlying OS is, and if the env privileges would allow us to start the docker daemon through shell. I might need your input here to understand how I should further proceed with this issue.
Thanks!

For context, here are the logs from the failed run:
```sh
499 make -C docs/book verify
500 make[1]: Entering directory '/home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/docs/book'
501 make -C ../ diagrams
502 make[2]: Entering directory '/home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/docs'
503 make -C ../hack/tools bin/plantuml-sentinal
504 make[3]: Entering directory '/home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/hack/tools'
505 docker build --build-arg PLANTUML_VERSION=1.2020.16  . -f plantuml.Dockerfile -t "plantuml-builder"
506 touch bin/plantuml-sentinal
507 failed to dial gRPC: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: dial unix /var/run/docker.sock: connect: no such file or directory
508 make[3]: *** [Makefile:161: bin/plantuml-sentinal] Error 1
509 make[3]: Leaving directory '/home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/hack/tools'
510 make[2]: *** [../common.mk:50: ../hack/tools/bin/plantuml-sentinal] Error 2
511 make[2]: Leaving directory '/home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/docs'
512 make[1]: *** [Makefile:61: generate] Error 2
513 make[1]: Leaving directory '/home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/docs/book'
514 make: *** [Makefile:620: verify-book-links] Error 2
```

**Comment by richardcase** [comment]
@sedefsavas @zeborg - as we want to run a container as part of the verify make target we need to run the prow job in privileged mode. This should do that:

https://github.com/kubernetes/test-infra/pull/25240

**Comment by sedefsavas** [comment]
> Our verify implementation in different to upstream capi (they just call mdbook build).

I think we can add that too. We only test uml creation as is.

**Comment by richardcase** [comment]
I just made a change to the prow job definition so that we execute the make target via runner.sh which setups DIND:

https://github.com/kubernetes/test-infra/pull/25264

**Comment by richardcase** [comment]
Looking more promising this time:
```
 Docker in Docker enabled, initializing...
================================================================================
Starting Docker: docker.
Waiting for docker to be ready, sleeping for 1 seconds. 
```

Fingers crossed

**Comment by richardcase** [comment]
Yay it passed

**Pull request merged**